src/field_theory/adjoin.lean

```diff
@@ -166,81 +259,150 @@ end adjoin_def
 section adjoin_subalgebra_lattice
 variables {F : Type*} [field F] {E : Type*} [field E] [algebra F E] {α : E} {S : set E}
 
-lemma adjoin_eq_bot (h : S ⊆ (⊥ : subalgebra F E)) : adjoin F S = ⊥ :=
-begin
-  rw eq_bot_iff,
-  intros x,
-  rw [subalgebra.mem_coe, subalgebra.mem_coe, algebra.mem_bot],
-  rw algebra.coe_bot at h,
-  apply adjoin_subset_subfield _ _ set.subset.rfl h,
-end
+lemma adjoin_eq_bot (h : S ⊆ (⊥ : intermediate_field F E)) : adjoin F S = ⊥ :=
+eq_bot_iff.mpr ((adjoin_le_iff F S).mpr h)
 
-lemma adjoin_simple_eq_bot (hα : α ∈ ((⊥ : subalgebra F E) : set E)) : F⟮α⟯ = (⊥ : subalgebra F E) :=
+lemma adjoin_simple_eq_bot (hα : α ∈ (⊥ : intermediate_field F E)) : F⟮α⟯ = ⊥ :=
 adjoin_eq_bot (set.singleton_subset_iff.mpr hα)
 
-lemma adjoin_zero : F⟮0⟯ = (⊥ : subalgebra F E) :=
-adjoin_simple_eq_bot (algebra.mem_bot.mpr (is_add_submonoid.zero_mem))
+lemma adjoin_zero : F⟮(0 : E)⟯ = ⊥ :=
+adjoin_simple_eq_bot (⊥ : intermediate_field F E).zero_mem
 
-lemma adjoin_one : F⟮1⟯ = (⊥ : subalgebra F E) :=
-adjoin_simple_eq_bot (algebra.mem_bot.mpr (is_submonoid.one_mem))
+lemma adjoin_one : F⟮(1 : E)⟯ = ⊥ :=
+adjoin_simple_eq_bot (⊥ : intermediate_field F E).one_mem
 
-lemma sub_bot_of_adjoin_sub_bot (h : adjoin F S = ⊥) : S ⊆ (⊥ : subalgebra F E) :=
+lemma sub_bot_of_adjoin_sub_bot (h : adjoin F S = ⊥) : S ⊆ (⊥ : intermediate_field F E) :=
 calc S ⊆ adjoin F S : subset_adjoin _ _
-  ... = (⊥ : subalgebra F E) : congr_arg coe h
+  ... = (⊥ : intermediate_field F E) : congr_arg coe h
 
-lemma mem_bot_of_adjoin_simple_sub_bot (h : F⟮α⟯ = ⊥) : α ∈ ((⊥ : subalgebra F E) : set E) :=
-set.singleton_subset_iff.mp (sub_bot_of_adjoin_sub_bot h)
+lemma mem_bot_of_adjoin_simple_sub_bot (h : F⟮α⟯ = ⊥) : α ∈ (⊥ : intermediate_field F E) :=
+  ((⊥ : intermediate_field F E).mem_coe α).mp
+    (set.singleton_subset_iff.mp (sub_bot_of_adjoin_sub_bot h))
 
-lemma adjoin_eq_bot_iff : S ⊆ (⊥ : subalgebra F E) ↔ adjoin F S = ⊥ :=
+lemma adjoin_eq_bot_iff : S ⊆ (⊥ : intermediate_field F E) ↔ adjoin F S = ⊥ :=
```

Comment:
The name suggests that the two sides should be swapped.
If you swap the two sides, it would also be a good simp lemma.